app_dart/lib/src/service/datastore.dart

```diff
@@ -208,12 +208,13 @@ class DatastoreService {
         head: pr.head.sha,
         status: null,
         updates: 0,
+        updateTimestamp: DateTime.now().millisecondsSinceEpoch,
       );
     } else {
       if (previousStatusUpdates.length > 1) {
-        throw StateError(
-            'GithubBuildStatusUpdate should have no more than one entries on '
-            'repository ${slug.fullName}, pr ${pr.number}, head ${pr.head.sha}');
+        return previousStatusUpdates.reduce((GithubBuildStatusUpdate current,
+                GithubBuildStatusUpdate next) =>
+            current.updateTimestamp < next.updateTimestamp ? next : current);
```

Comment:
It would be less confusing to directly query the latest record, e.g. adding `.order('-timestamp').limit(1)` to the query.

Reply:
It is a good point, and I have thought about this as well. 

To enable `order().limt()`, a complex composite index is needed. From the query, there are already three filter options there. Building that index is heavy, and it doesn't cove old data. Additionally it will be rarely used if we add that index (duplicate cases happen rarely).

On the other hand, we have been trying to get rid of composite indexes as much as possible to lower datastore traffic.

So I have decided to use `reduce` instead, and it is only called when the rare duplicate cases occure.

Added some doc to make it clear.